packages/contract-dependencies-gnosis/src/ContractDependenciesGnosis.ts

```diff
@@ -136,25 +170,51 @@ export class ContractDependenciesGnosis extends ContractDependenciesEthers {
       return super.sendTransaction(tx, txMetadata);
     }
     let txHash: string;
+
     // If the Relay Service is not being used so we'll execute the TX directly
     const prevTransaction = await this._currentSignRequest;
     const relayTransaction = await this.ethersTransactionToRelayTransaction(tx);
 
-    if (prevTransaction === relayTransaction)
+    if (prevTransaction === relayTransaction) {
       throw new Error('Message signature failed.');
+    }
 
     if (this.useRelay) {
-      txHash = await this.gnosisRelay.execTransaction(relayTransaction);
+      try {
+        txHash = await this.gnosisRelay.execTransaction(relayTransaction);
+      } catch (error) {
+        this.setUseRelay(false);
+        throw TransactionStatus.RELAYER_DOWN;
+      }
     } else {
       txHash = await this.execTransactionDirectly(relayTransaction);
     }
 
-    this.onTransactionStatusChanged(
-      txMetadata,
-      TransactionStatus.PENDING,
-      txHash
-    );
-    const response = await this.provider.getTransaction(txHash);
+    let response = await this.provider.getTransaction(txHash);
+
+    if (response) {
+      this.onTransactionStatusChanged(
+        txMetadata,
+        TransactionStatus.PENDING,
+        txHash
+      );
+
+      return response.wait();
+    }
+
+    // TODO - current worrkaround / explore a better fix
+    // loop getTransaction calls until we get back a response
+    for (let i = 0; i >= 0; i++) {
```

Comment:
@andrewdamelio Did we ever figure out why we had to do this?

Reply:
@justinbarry just re-tested this and it doesn't seem to be needed anymore. i'm going to remove until it becomes a problem.